cards/src/main/resources/cards/custom/group3/bug/kobolds_and_catacombs/secret_obvious_ambush.json

```diff
@@ -1,33 +1,27 @@
 {
-  "name": "Obvious Ambush",
+  "name": "Call the Cavalry",
   "baseManaCost": 1,
   "type": "SPELL",
   "heroClass": "AMBER",
   "rarity": "EPIC",
-  "description": "Secret: When an enemy minion attacks, summon a 3/3 Worm",
-  "secret": {
-    "class": "TargetAcquisitionTrigger",
-    "actionType": "PHYSICAL_ATTACK",
-    "fireCondition": {
-      "class": "MinionCountCondition",
-      "value": 7,
-      "operation": "LESS"
-    },
-    "queueCondition": {
-      "class": "MinionCountCondition",
-      "value": 7,
-      "operation": "LESS"
-    },
-    "sourceEntityType": "MINION",
-    "sourcePlayer": "OPPONENT",
-    "targetPlayer": "BOTH"
-  },
+  "description": "At the end of this turn, summon a 3/3 Worm",
   "spell": {
-    "class": "SummonSpell",
-    "value": 1,
-    "card": "token_shiny_worm"
-  },
+    "class": "AddEnchantmentSpell",
+    "target": "FRIENDLY_PLAYER",
+    "trigger": {
+      "eventTrigger": {
+        "class": "TurnEndTrigger",
+        "targetPlayer": "SELF"
+      },
+      "spell": {
+        "class": "SummonSpell",
+        "value": 1,
+        "card": "token_shiny_worm"
+      },
+      "oneTurn": true
```

Comment:
maybe this would be more appropriate as a `"maxFires": 1`.

Also, does `maxFires` interact with double turn end triggers? I don't actually know! My feeling is that it does not.

Reply:
The DoubleTurnEndTriggers is listed #1274 

I think we should use oneTurn for these sorts of effects precisely so that they will synergize with the fixed effect, since I believe maxFires will not.

Reply:
did we fix double end turn triggers everywhere?

Reply:
(still needs to be addressed)

Reply:
Now addressed and tested